File: Java/1_Two_Sum.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> result = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (result.containsKey(complement)) {
                return new int[]{result.get(complement), i};
            } else {
                result.put(nums[i], i);
            }
        }
        return null;
    }
}

File: C#/1_Two_Sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> result = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++) {
            int complement = target - nums[i];
            if (result.ContainsKey(complement)) {
                return new [] { result[complement], i };
            } else {
                result[nums[i]] = i;
            }
        }
        return null;
    }
}
